commerce-frontend-taglib/src/main/java/com/liferay/commerce/frontend/taglib/servlet/taglib/ProductSubscriptionInfoTag.java

```diff
@@ -0,0 +1,191 @@
+
+			_duration = _length * cpSubscriptionInfo.getMaxSubscriptionCycles();
+
+			String subscriptionType = cpSubscriptionInfo.getSubscriptionType();
+
+			String period = StringPool.BLANK;
+
+			ThemeDisplay themeDisplay = (ThemeDisplay)request.getAttribute(
+				WebKeys.THEME_DISPLAY);
+
+			CPSubscriptionType cpSubscriptionType =
+				cpSubscriptionTypeRegistry.getCPSubscriptionType(
+					subscriptionType);
+
+			if (cpSubscriptionType != null) {
+				period = cpSubscriptionType.getLabel(themeDisplay.getLocale());
+			}
+
+			_subscriptionPeriodKey = _getPeriodKey(period, _length != 1);
+
+			_durationPeriodKey = _getPeriodKey(period, _duration != 1);
+		}
+		catch (PortalException pe) {
+			if (_log.isDebugEnabled()) {
+				_log.debug(pe, pe);
+			}
+
+			return SKIP_BODY;
+		}
+
+		String durationPeriod = StringPool.BLANK;
+		String subscriptionPeriod = StringPool.BLANK;
+
+		if (_showDuration && (_duration > 0)) {
+			durationPeriod = LanguageUtil.format(
+				request, "duration-x-x",
+				new Object[] {_duration, _durationPeriodKey});
+		}
+
+		if ((_length > 0) && Validator.isNotNull(_subscriptionPeriodKey)) {
+			subscriptionPeriod = LanguageUtil.format(
+				request, "every-x-x",
+				new Object[] {_length, _subscriptionPeriodKey});
+		}
+
+		if (Validator.isNotNull(durationPeriod)) {
+			putValue("durationPeriod", durationPeriod);
+		}
+
+		if (Validator.isNotNull(subscriptionPeriod)) {
+			putValue("subscriptionPeriod", subscriptionPeriod);
+		}
+
+		setTemplateNamespace("SubscriptionInfo.render");
+
+		return super.doStartTag();
+	}
+
+	public long getCPInstanceId() {
+		return _cpInstanceId;
+	}
+
+	@Override
+	public String getModule() {
+		NPMResolver npmResolver = ServletContextUtil.getNPMResolver();
+
+		if (npmResolver == null) {
+			return StringPool.BLANK;
+		}
+
+		return npmResolver.resolveModuleName(
+			"commerce-frontend-taglib/js/subscription_info" +
+				"/SubscriptionInfo.es");
+	}
+
+	public boolean isShowDuration() {
+		return _showDuration;
+	}
+
+	public void setCPInstanceId(long cpInstanceId) {
+		_cpInstanceId = cpInstanceId;
+	}
+
+	@Override
+	public void setPageContext(PageContext pageContext) {
+		super.setPageContext(pageContext);
+
+		cpSubscriptionTypeRegistry =
+			ServletContextUtil.getCPSubscriptionTypeRegistry();
+		servletContext = ServletContextUtil.getServletContext();
+	}
+
+	public void setShowDuration(boolean showDuration) {
+		_showDuration = showDuration;
+	}
+
+	@Override
+	protected void cleanUp() {
+		super.cleanUp();
+
+		_cpInstanceId = 0;
+		_duration = 0;
+		_durationPeriodKey = null;
+		_length = 0;
+		_showDuration = true;
+		_subscriptionPeriodKey = null;
+	}
+
+	protected CPSubscriptionTypeRegistry cpSubscriptionTypeRegistry;
+
+	private String _getPeriodKey(String period, boolean plural) {
+		if (plural) {
+			return LanguageUtil.get(
+				request,
+				StringUtil.toLowerCase(period) + CharPool.LOWER_CASE_S);
+		}
+
+		return period;
```

Comment:
Why do we localize the `period` if `plural` is true, but we don't localize if `plural` is false? Seems like we should do it both times or neither time.